Update draft-ietf-rats-architecture.md (#86)
ietf-rats-wg/architecture: draft-ietf-rats-architecture.md

RATS are facilitated by an additional vital party, the Verifier.
The Verifier appraises Evidence via Appraisal Policies and creates the Attestation Results to support Relying Parties in their decision process.

> This documents defines a flexible architecture consisting of attestation Roles and their interactions via Role Messages.

Reviewer: Change "Role Messages" back to "conceptual messages".  The latter is the name of the section title, and we don't need to define a new term.
```suggestion
This documents defines a flexible architecture consisting of attestation Roles and their interactions via conceptual messages.
```

> This documents defines a flexible architecture consisting of attestation Roles and their interactions via conceptual messages.

Reviewer: Can we not capitalize Roles?  Should not need to define a term for Roles, just use it.

> This documents defines a flexible architecture consisting of attestation Roles and their interactions via conceptual messages.

Reviewer 2: ```suggestion
This documents defines a flexible architecture consisting of attestation roles and their interactions via conceptual messages.
```

Additionally, this document defines a universal set of terms that can be mapped to various existing and emerging Remote Attestation Procedures.
Common topological models and the data flows associated with them, such as the "Passport Model" and the "Background-Check Model" are illustrated.
The purpose is to enable readers to map their solution architecture to the
canonical attestation architecture provided here and to define useful terminology for attestation.
Having a common terminology that provides well-understood meanings for common themes

> such as, Roles, device composition, Topological Models and appraisal is vital for

Reviewer: ```suggestion
such as, roles, device composition, topological models and appraisal is vital for
```
use capitalization consistent with line 94 which has "topological models"

> such as, roles, device composition, topological models and appraisal is vital for

Reviewer: ```suggestion
such as roles, device composition, topological models, and appraisal is vital for
```

Attestation Result:

: The output generated by a Verifier, typically including information about an Attester, where the Verifier vouches for the validity of the Evidence it has appraised

Attester:

> : An entity (typically a device), whose Evidence must be appraised in order to determine whether it is considered trustworthy, such as when deciding whether it is authorized to perform some operation

Reviewer: I don't think "typically a device" is accurate.  In TEEP it's certainly not.   In SGX in general, it's not.   Better to be silent and not say what is typical.

Reviewer: "the extent to which it is trustworthy"

> : An entity (typically a device), whose Evidence must be appraised in order to determine whether it is considered trustworthy, such as when deciding whether it is authorized to perform some operation

Reviewer: Is "it" referring to "Evidence" now?  That doesn't seem right.   Revert changes to "it".

Reviewer: or "the Attester"

Endorsement:

> : Statements that Endorsers make (typically a manufacturer) that vouches for the design and implementation of the Attester. Often this includes statements about the integrity of an Attester's signing capability

Reviewer: Bad grammar now: "Statements .. vouches" doesn't match.

> : Statements that Endorsers make (typically a manufacturer) that vouches for the design and implementation of the Attester. Often this includes statements about the integrity of an Attester's signing capability

Reviewer 2: ```suggestion
: Statements that Endorsers make (typically a manufacturer) that vouch for the design and implementation of the Attester. Often this includes statements about the integrity of an Attester's signing capability
```

Endorser:

: An entity (typically a manufacturer) whose Endorsements help Verifiers appraise the authenticity of Evidence

Evidence:

> : A set of information that asserts the trustworthiness status of an Attester, that is appraised by a Verifier

Reviewer: Grammar of the last phrase is now harder to read in my view.

> : A set of information that asserts the trustworthiness status of an Attester, that is appraised by a Verifier

Reviewer 2: ```suggestion
: A set of claims created by the Attester that characterize the Attestation Target(s). The Attestation Evidence is conveyed from the Attester to the Verifier where it is appraised.
```

Relying Party:

> : An entity, that depends on the validity of information about an Attester, for purposes of reliably applying application specific actions.  Compare /relying party/ in {{RFC4949}}

Reviewer 2: ```suggestion
: An entity that depends on the validity of information about an Attester. Compare /relying party/ in {{RFC4949}}.
```

> 

Reviewer: Discussion during meeting: no consensus on the phrase "typically an application" being useful

> 

Reviewer: Another possibility:

An entity that depends on the validity of information about an Attester.  Compare /relying party/ in {{RFC4949}}.

> 

Author: +1 to Dave's suggested wording.

Relying Party Owner:

: An entity (typically an administrator), that is authorized to configure Appraisal Policy for Attestation Results in a Relying Party

Verifier:

> : An entity (typically a service provider), that appraises the validity of Evidence about an Attester

Reviewer: I don't think we should introduce the term "service provider" here, we'd have to define it.

Reviewer: Laurence suggests "service" instead of "service provider"

~~~~
{: #combination title="Example Combination"}

# Role Hosting and Composition

> The RATS architecture includes the definition of Roles (e.g. Attester, Verifier,
> Relying Party, Endorser) and Role Messages (e.g. Evidence, Attestation Results,

Reviewer: ```suggestion
The RATS architecture includes the definition of Roles (e.g., Attester, Verifier,
Relying Party, Endorser) and conceptual messages (e.g., Evidence, Attestation Results,
```

Endorsements, Appraisal Policies) that captures canonical attestation behaviors,
that are common to a broad range of attestation-enabled systems. An entity that

> combines multiple Roles produces and consumes the associated Role Messages.

Reviewer: I do not believe we should invent/define a term "Role Messages".

Nevertheless, the semantics of which Role Messages are inputs and outputs to a Role
entity remains constant. As a result, the entity can participate as a constituent of the RATS architecture
while flexibly accommodating the needs of various deployment architectures.

Interactions between Roles do not necessarily require use of Internet protocols.
They could, for example, use inter-process communication, local system buses,
shared memory, hypervisors, IP-loopback devices or any communication path
between the various environments that may exist on the entity that combines multiple Roles.

The movement of Role Messages between locally hosted Roles is referred to as
"local conveyance". Most importantly, the definition of local conveyance methods is
out-of-scope for the RATS architecture.

The following paragraph elaborates on an exemplary usage scenario:

In a Composite Device scenario, in addition to local entities that host the
lead Attester and other subordinate Attesters, the Composite Device can host the
Verifier role locally to appraise Evidence from one or more subordinate Attesters.
The local Verifier might convey local Attestation Results to a remote Relying party

> or the Relying Party role also could become local where an application specific

Reviewer: ```suggestion
or the Relying Party role also could become local where an application-specific
```

action is taken locally. For example, a secure boot scenario prevents system software

> from loading if the firmware fails to satisfy a local trustworthiness appraisal policy.

Reviewer: ```suggestion
from loading if the firmware fails to satisfy a local appraisal policy.
```

In a multi-network scenario, a network node might bridge a wide-area

> network, local-area network and span various system buses. In so doing, the bridge node 

Reviewer: ```suggestion
network, local-area network, and span various system buses. In so doing, the bridge node 
```

might need to host multiple Roles depending on the type of behavior each connected domain
expects. For example, the node might be an Attester to a wide-area network, a Verifier to

> the local-area network and a Relying Party to components attached to a local system bus.

Reviewer: ```suggestion
the local-area network, and a Relying Party to components attached to a local system bus.
```
